spec/models/stripe_account_spec.rb

```diff
@@ -4,28 +4,30 @@
 require 'stripe/oauth'
 
 describe StripeAccount do
-  describe "deauthorize_and_destroy" do
+  describe "deauthorize_and_destroy", :vcr, :stripe_version do
     let!(:enterprise) { create(:enterprise) }
     let!(:enterprise2) { create(:enterprise) }
-    let(:client_id) { 'ca_abc123' }
-    let(:stripe_user_id) { 'acct_abc123' }
+    let(:client_id) { client.id }
+    let(:stripe_user_id) { ENV.fetch('STRIPE_ACCOUNT', nil) }
+    let(:client) do
+      Stripe::Customer.create({
+                                name: 'Bumble Bee',
+                                email: 'bumblebee@example.com',
+                              })
+    end
+
     let!(:stripe_account) {
       create(:stripe_account, enterprise:, stripe_user_id:)
     }
 
+    let(:secret) { ENV.fetch('STRIPE_SECRET_TEST_API_KEY', nil) }
+
     before do
-      Stripe.api_key = "sk_test_12345"
+      Stripe.api_key = secret
       Stripe.client_id = client_id
     end
 
     context "when the Stripe API disconnect fails" do
-      before do
-        stub_request(:post, "https://connect.stripe.com/oauth/deauthorize").
-          with(body: { "client_id" => client_id, "stripe_user_id" => stripe_user_id }).
-          to_return(status: 400, body: JSON.generate(error: 'invalid_grant',
-                                                     error_description: "Some Message"))
-      end
-
```

Comment:
I wonder if this stub was correct in the first place, or rather the test setup is not appropriate. From the real Stripe call, we get the response:

- with a `status: 200`
- with  
```
body:
      encoding: UTF-8
      string: |-
        {
          "error": "invalid_client",
          "error_description": "No such application: 'cus_PDCEEWCCEIDICb'"
        }
```

A status 400 would be a bad request?...